kafka/producer/base.py

```diff
@@ -108,15 +113,23 @@ class Producer(object):
     def __init__(self, client, async=False,
                  req_acks=ACK_AFTER_LOCAL_WRITE,
                  ack_timeout=DEFAULT_ACK_TIMEOUT,
-                 codec=None,
+                 codec=CODEC_NONE,
                  batch_send=False,
                  batch_send_every_n=BATCH_SEND_MSG_COUNT,
-                 batch_send_every_t=BATCH_SEND_DEFAULT_INTERVAL):
+                 batch_send_every_t=BATCH_SEND_DEFAULT_INTERVAL,
+                 maxsize=ASYNC_QUEUE_MAXSIZE):
```

Comment:
Would put `maxsize=None` here. Then in the body below, you can do:

    if maxsize is None:
        maxsize = ASYNC_QUEUE_MAXSIZE

That allows `SimpleProducer` to pass in `None` and get the default behaviour.